### PR TITLE
Require that emails on GitHub & Generic OAuth2 principals be verified & primary, if those fields are provided

### PR DESCRIPTION
**Pull request by jsternberg**
If an account had multiple emails, the current implementation would
always select the first one regardless of any other settings. This fixes
it so it only chooses the primary email address that is verified.

### CODE REVIEW

File: oauth2/github.go [hunk 1]
```diff
@@ -143,7 +144,7 @@ func getOrganizations(client *github.Client, log chronograf.Logger) ([]*github.O
 			PerPage: 10,
 		}
 		// Get the organizations for the current authenticated user.
-		orgs, resp, err := client.Organizations.List("", opt)
+		orgs, resp, err := client.Organizations.List(context.Background(), "", opt)
```
Comment (goller) [inline oauth2/github.go]: This should be context.TODO() until context is wire through via
https://github.com/influxdata/chronograf/blob/0d639d5177b72c12d9470e10ea69ff2dfff45a94/oauth2/oauth2.go#L55

File: oauth2/github.go [hunk 2]
```diff
@@ -159,7 +160,7 @@ func getOrganizations(client *github.Client, log chronograf.Logger) ([]*github.O
 
 // getPrimaryEmail gets the primary email account for the authenticated user.
 func getPrimaryEmail(client *github.Client, log chronograf.Logger) (string, error) {
-	emails, resp, err := client.Users.ListEmails(nil)
+	emails, resp, err := client.Users.ListEmails(context.Background(), nil)
```
Comment (goller) [inline oauth2/github.go]: Also `context.TODO()`

**Comment by goller** [comment]
Hey @jsternberg Would you add a test for a different email here:

https://github.com/influxdata/chronograf/blob/73779344581f96974ffade003ded40b069c13fb4/oauth2/github_test.go#L13

**Comment by jsternberg** [comment]
@goller fixed.

**Comment by jsternberg** [comment]
The tests for generic seem to not set verified for a bunch of them. Should I modify it so that verified is required there or make it so verified is required only if it is present? Since it's the generic oauth2 implementation, I don't necessarily know what is expected.

**Comment by russorat** [comment]
@jsternberg i can't find in the OAuth2 spec that verified is required so i think it should only be required if it is present.

**Comment by jaredscheib** [comment]
From what I can tell, there is no standard for verified emails in a generic OAuth2 implementation. OpenID Connect apparently provides a claim for this, but we allow any providers to be set up. So let's go with "required only if present".

**Comment by jsternberg** [comment]
I've updated the PR and also reverted the vendoring change.

**Pull request merged**